feat: Support for IF EXISTS on CREATE CONNECTOR (#6036)
confluentinc/ksql: docs/developer-guide/ksqldb-reference/create-connector.md

--------

```sql

> CREATE SOURCE | SINK CONNECTOR connector_name [IF EXISTS] WITH( property_name = expression [, ...]);

Reviewer: ```suggestion
CREATE SOURCE | SINK CONNECTOR connector_name [IF NOT EXISTS] WITH( property_name = expression [, ...]);

```

Author: Ahh ! This slipped right through , fixed this

> CREATE SOURCE | SINK CONNECTOR connector_name [IF NOT EXISTS] WITH( property_name = expression [, ...]);

Reviewer 2: ```suggestion
CREATE SOURCE | SINK CONNECTOR [IF NOT EXISTS] connector_name WITH( property_name = expression [, ...]);

```

Author: fixed this

that simplify configuring them. For more information, see
[Natively Supported Connectors](../../concepts/connectors.md#natively-supported-connectors).

> If the IF EXISTS clause is present, the statement doesn't fail if the
> type doesn't exist.
> 

Reviewer: Note sure why this talks about _types_ and them _not_ existing. Isn't this about _connectors_ that _do_ exist?

```suggestion
If the `IF NOT EXISTS` clause is present, the statement does nothing if a connector with the supplied name already exists.

```

??

Author: Yes , I have changed this , according to the behavior similar `CREATE STREAMS`  though there was no docs for it

> If the IF NOT EXISTS clause is present, the statement will fail if a connector with the supplied name

Reviewer 2: Contrary, if the `IF NOT EXISTS` clause is present, the statement doesn't fail if the connector already exists.

Author: @spena I see , the current implementation in this PR throws an exception if a connector exists ,
so the expectation would be to simply pass along an `ErrorEntity` and skipping the connector creation part of the code ?

Reviewer 2: Yes. 

Btw, there's another PR to support the DROP CONNECTOR IF EXISTS and included a `WarningEntity` class. It would be good to return that instead of the `ErrorEntity` (See https://github.com/confluentinc/ksql/pull/6067/files#diff-748287344db7330d94327cc33a1117e9)
